tracer/src/Datadog.Trace.MSBuild/DatadogLogger.cs

```diff
@@ -337,15 +343,65 @@ private void EventSource_WarningRaised(object sender, BuildWarningEventArgs e)
                         }
                     }
 
-                    LogItem logItem = new LogItem("warn", message, type, code, lineNumber, columnNumber, endLineNumber, endColumnNumber, projectFile, filePath, stack, subCategory);
-                    string logMessage = correlation + JsonConvert.SerializeObject(logItem);
-                    Console.WriteLine(logMessage);
+                    _tracer.TracerManager.DirectLogSubmission.Sink.EnqueueLog(new MsBuildLogEvent(e.Message, projectSpan));
                 }
             }
             catch (Exception ex)
             {
                 Log.Error(ex, "Error in WarningRaised event");
             }
         }
+
+        private class MsBuildLogEvent : DatadogLogEvent
+        {
+            private readonly string _message;
+            private readonly Context? _context;
+
+            public MsBuildLogEvent(string message, Span span)
+            {
+                _message = message;
+                _context = span is null ? null : new Context(span.TraceId, span.SpanId, span.Context.Origin);
+            }
+
+            public override void Format(StringBuilder sb, LogFormatter formatter)
+            {
+                formatter.FormatLog<Context?>(
+                    sb,
+                    _context,
```

Comment:
```suggestion
                    in _context,
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/2894/commits/96b5459bf6e21942189da394d10232ad68963d81